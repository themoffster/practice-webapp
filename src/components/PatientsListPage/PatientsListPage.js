import { patientPropTypes } from 'src/enums/propsTypes'
import { Table } from 'react-bootstrap'
import PropTypes from 'prop-types'
import React from 'react'

const PatientsListPage = ({ patients }) => {
  return (
    <Table striped bordered hover>
      <thead>
        <tr>
          <th>Name</th>
          <th>DOB</th>
        </tr>
      </thead>
      <tbody>
        {patients.map((patient) => (
          <tr key={patient.id}>
            <td>
              {patient.firstname} {patient.lastname}
            </td>
            <td>{patient.dob}</td>
          </tr>
        ))}
      </tbody>
    </Table>
  )
}

PatientsListPage.defaultProps = {
  patients: [],
}

PatientsListPage.propTypes = {
  patients: PropTypes.arrayOf(patientPropTypes).isRequired,
}

export default PatientsListPage
