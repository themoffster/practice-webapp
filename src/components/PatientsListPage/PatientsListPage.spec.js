import { render } from '@testing-library/react'
import PatientsListPage from './PatientsListPage'
import React from 'react'

const getWrapper = (props) => render(<PatientsListPage {...props} />)

describe('PatientsListPage', () => {
  it('matches the snapshot', () => {
    const props = {
      patients: [],
    }
    const { getFragment } = getWrapper(props)
    expect(getFragment).toMatchSnapshot()
  })
})
