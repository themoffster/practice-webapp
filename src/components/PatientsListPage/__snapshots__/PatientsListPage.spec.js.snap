// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PatientsListPage matches the snapshot 1`] = `undefined`;
